### PR TITLE
Add "show all" in auto-completion

### PR DESCRIPTION
**Pull request by Zhiying12**
When using auto-completion, the first row of the result list is "show all" whose value is a path to show all customers who contain the keyword in their information. The only methods modified is auto_complete_for_customer in customer controller.

### CODE REVIEW

File: app/controllers/customers_controller.rb [hunk 1]
```diff
@@ -279,8 +279,11 @@ def auto_complete_for_customer
     @customers_s =
       Customer.find_by_multiple_terms(s.split( /\s+/)).
           reject {|customer| @customers.include?(customer)}
-    result = @customers.map do |c|
-      {'label' => c.full_name, 'value' => customer_path(c)}
+    session[:customers_filter] = params[:term]
+    result = Array[{'label' => 'show all',
```
Comment (fxdawnn) [inline app/controllers/customers_controller.rb:283]: Is it duplicate to use Array() and [ ] to initialize at the same time?

File: features/customers/search_with_autocomplete.feature [hunk 2]
```diff
@@ -34,7 +35,8 @@ Scenario:search with other information
     | Organ      | Milk      | dancingfox@mail.com | 100 bway      |  SAF | CA    |
 
   When I fill "search_field" autocomplete field with "Fox"
-  Then I should see autocomplete choice "Armando Fox"
+  Then I should see autocomplete choice "show all"
```
Comment (fxdawnn) [inline features/customers/search_with_autocomplete.feature:38]: It is not tested that when pressing "show all", we would go to the page of everything.

**Comment by fxdawnn** [comment]
@WayneLeung12  PTAL at comments.

File: app/controllers/customers_controller.rb (hunk 1 above)
Comment (Zhiying12) [reply app/controllers/customers_controller.rb:283]: From my perspective, they are the same.

File: features/customers/search_with_autocomplete.feature (hunk 2 above)
Comment (Zhiying12) [reply features/customers/search_with_autocomplete.feature:38]: You're right, I missed this test case.

**Comment by jayl109** [comment]
LGTM

**Pull request merged**